arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -807,7 +808,14 @@ Result fromFuncGeoInRange(char const* funcName, irs::boolean_filter* filter,
   if (!nameFromAttributeAccess(name, *fieldNode, ctx, filter != nullptr)) {
     return error::failedToGenerateName(funcName, fieldNodeIdx);
   }
-  if (filter) {
+
+  tmpValue.reset(*centroidNode);
+  if (buildFilter || tmpValue.isConstant()) {
```

Comment:
We could move that inside 'if (buildFilter)'

Reply:
Done